exporter/datadogexporter/internal/logs/translator_test.go

```diff
@@ -0,0 +1,285 @@
+				},
+			},
+		},
+		{
+			name: "log_with_trace",
+			args: args{
+				lr: func() plog.LogRecord {
+					l := plog.NewLogRecord()
+					l.Attributes().InsertString("app", "test")
+					l.SetSpanID(spanID)
+					l.SetTraceID(traceID)
+					l.Attributes().InsertString(conventions.AttributeServiceName, "otlp_col")
+					l.SetSeverityNumber(5)
+					return l
+				}(),
+				res: func() pcommon.Resource {
+					r := pcommon.NewResource()
+					return r
+				}(),
+			},
+			want: datadogV2.HTTPLogItem{
+				Message: *datadog.PtrString(""),
+				Service: datadog.PtrString("otlp_col"),
+				AdditionalProperties: map[string]string{
+					"app":              "test",
+					"status":           "debug",
+					otelSeverityNumber: "5",
+					otelSpanID:         fmt.Sprintf("%x", string(spanID[:])),
+					otelTraceID:        fmt.Sprintf("%x", string(traceID[:])),
+					ddSpanID:           fmt.Sprintf("%d", ddSp),
+					ddTraceID:          fmt.Sprintf("%d", ddTr),
+
+					"service.name": "otlp_col",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := Transform(tt.args.lr, tt.args.res)
+
+			gs, err := got.MarshalJSON()
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+			ws, err := tt.want.MarshalJSON()
+			if err != nil {
+				t.Fatal(err)
+				return
+			}
+			if !assert.JSONEq(t, string(ws), string(gs)) {
+				t.Errorf("Transform() = %v, want %v", string(gs), string(ws))
+			}
+		})
+	}
+}
+
+func Test_deriveStatus(t *testing.T) {
+	type args struct {
+		severity plog.SeverityNumber
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "trace3",
+			args: args{
+				severity: 3,
+			},
+			want: logLevelTrace,
+		},
+		{
+			name: "trace4",
+			args: args{
+				severity: 4,
+			},
+			want: logLevelTrace,
+		},
+		{
+			name: "debug5",
+			args: args{
+				severity: 5,
+			},
+			want: logLevelDebug,
+		},
+		{
+			name: "debug7",
+			args: args{
+				severity: 7,
+			},
+			want: logLevelDebug,
+		},
+		{
+			name: "debug8",
+			args: args{
+				severity: 8,
+			},
+			want: logLevelDebug,
+		},
+		{
+			name: "info9",
+			args: args{
+				severity: 9,
+			},
+			want: logLevelInfo,
+		},
+		{
+			name: "info12",
+			args: args{
+				severity: 12,
+			},
+			want: logLevelInfo,
+		},
+		{
+			name: "warn13",
+			args: args{
+				severity: 13,
+			},
+			want: logLevelWarn,
+		},
+		{
+			name: "warn16",
+			args: args{
+				severity: 16,
+			},
+			want: logLevelWarn,
+		},
+		{
+			name: "error17",
+			args: args{
+				severity: 17,
+			},
+			want: logLevelError,
+		},
+		{
+			name: "error20",
+			args: args{
+				severity: 20,
+			},
+			want: logLevelError,
+		},
+		{
+			name: "fatal21",
+			args: args{
+				severity: 21,
+			},
+			want: logLevelFatal,
+		},
+		{
+			name: "fatal24",
+			args: args{
+				severity: 24,
+			},
+			want: logLevelFatal,
+		},
+		{
+			name: "undefined",
+			args: args{
+				severity: 50,
+			},
+			want: logLevelError,
```

Comment:
This is maybe just a style nit but I think this test might be simpler if we just wrote out a separate assertion for each case, rather than looping through this ~100 line map.

E.g. something like
```
...
assert.Equalf(t, logLevelError, derviveDdStatusFromSeverityNumber(21)
assert.Equalf(t, logLevelFatal, derviveDdStatusFromSeverityNumber(24)
...
```

Reply:
Then again maybe you prefer it this way to keep consistent with other tests 🤷‍♂️

Reply:
I would prefer to have same format for all tests.